src/libcore/lib.rs

```diff
@@ -177,3 +185,21 @@ mod char_private;
 mod iter_private;
 mod tuple;
 mod unit;
+
+// Pull in the the `coresimd` crate directly into libcore. This is where all the
+// architecture-specific (and vendor-specific) intrinsics are defined. AKA
+// things like SIMD and such. Note that the actual source for all this lies in a
+// different repository, rust-lang-nursery/stdsimd. That's why the setup here is
+// a bit wonky.
+#[path = "../stdsimd/coresimd/mod.rs"]
+#[allow(missing_docs, missing_debug_implementations, dead_code)]
+#[unstable(feature = "stdsimd", issue = "0")]
+#[cfg(not(stage0))] // allow changes to how stdsimd works in stage0
+mod coresimd;
```

Comment:
Silly question: does this stage0 check preclude core/std from using anything from stdsimd?

Reply:
It would during stage0, yeah, but I'd imagine that most usage would largely be an optimization in which case tagging it as `#[cfg(not(stage0))]` probably wouldn't be too bad.

Reply:
Note that the alternative here is to manually add `#[cfg(stage0)]` tags and such to stdsimd itself rather than blanket defining it for this module, and I figured that for the time being it's easiest to take this route which is only defining it in stage1+

Reply:
Oh, good point.  Yeah, the stage1 compiler just being a bit slower in some corner cases than it could be probably isn't a big deal.